src/app/server/OnboardingCodesUtil.cpp

```diff
@@ -113,7 +115,8 @@ void ShareQRCodeOverNFC(chip::RendezvousInformationFlags aRendezvousFlags)
 
 CHIP_ERROR GetSetupPayload(chip::SetupPayload & aSetupPayload, chip::RendezvousInformationFlags aRendezvousFlags)
 {
-    CHIP_ERROR err                      = CHIP_NO_ERROR;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+#if CONFIG_DEVICE_LAYER
```

Comment:
Why not refactor and raise issues to avoid a dependency that seems wrong, and breaks the example you are working on, rather than adding more #ifdefs that just push the problem further down?

Reply:
No changes to OnboardingCodesUtil or other classes under app/server in the updated diff.